nucypher/characters/lawful.py

```diff
@@ -1102,71 +1097,65 @@ def __attempt(attempt=1, interval=10) -> Ursula:
                                  network_middleware: RestMiddleware = None,
                                  *args,
                                  **kwargs
                                  ) -> 'Ursula':
 
+        if network_middleware is None:
+            network_middleware = RestMiddleware()
+
         #
         # WARNING: xxx Poison xxx
         # Let's learn what we can about the ... "seednode".
         #
 
-        if network_middleware is None:
-            network_middleware = RestMiddleware()
-
+        # Parse node URI
         host, port, checksum_address = parse_node_uri(seed_uri)
 
         # Fetch the hosts TLS certificate and read the common name
         certificate = network_middleware.get_certificate(host=host, port=port)
         real_host = certificate.subject.get_attributes_for_oid(NameOID.COMMON_NAME)[0].value
+
+        # Create a temporary certificate storage area
         temp_node_storage = ForgetfulNodeStorage(federated_only=federated_only)
-        certificate_filepath = temp_node_storage.store_node_certificate(certificate=certificate)
+        temp_certificate_filepath = temp_node_storage.store_node_certificate(certificate=certificate)
+
         # Load the host as a potential seed node
         potential_seed_node = cls.from_rest_url(
+            blockchain=blockchain,
             host=real_host,
             port=port,
             network_middleware=network_middleware,
-            certificate_filepath=certificate_filepath,
+            certificate_filepath=temp_certificate_filepath,
             federated_only=federated_only,
             *args,
-            **kwargs)  # TODO: 466
-
-        potential_seed_node.certificate_filepath = certificate_filepath
-
-        if checksum_address:
-            # Ensure this is the specific node we expected
-            if not checksum_address == potential_seed_node.checksum_address:
-                template = "This seed node has a different wallet address: {} (expected {}). " \
-                           " Are you sure this is a seednode?"
-                raise potential_seed_node.SuspiciousActivity(
-                    template.format(potential_seed_node.checksum_address,
-                                    checksum_address))
+            **kwargs
+        )
 
         # Check the node's stake (optional)
-        if minimum_stake > 0:
-            # TODO: check the blockchain to verify that address has more then minimum_stake. #511
-            raise NotImplementedError("Stake checking is not implemented yet.")
+        if minimum_stake > 0 and not federated_only:
+            staking_agent = StakingEscrowAgent(blockchain=blockchain)
+            seednode_stake = staking_agent.get_locked_tokens(staker_address=checksum_address)
+            if seednode_stake < minimum_stake:
+                raise Learner.NotATeacher(f"{checksum_address} is staking less then the specified minimum stake value ({minimum_stake}).")
```

Comment:
Be aware that checking whether this seednode is staking is already done in the next lines with `potential_seed_node.verify_node()`, which ends up calling `potential_seed_node._staker_is_really_staking()`.

This check would only be useful when `minimum_stake` is higher than the default minimum stake (currently 0 but subject to change as per https://github.com/nucypher/nucypher/issues/1115). In what use cases we'd want to strengthen the default minimum stake? If there are none, then perhaps we should just get rid of this parameter in this method.

Another way to see is: are Teachers inherently different from every other node to be required to have a higher stake?